src/decode.rs

```diff
@@ -165,21 +167,28 @@ pub fn decode_html_rw<R: BufRead, W: Write>(
             Numeric if c == 'x' => state = Hex,
             Dec if c == ';' => {
                 state = Normal;
-                let ch = try_parse!(decode_numeric(&buf[1..], 10), good_pos);
+                let ch = try_parse!(
+                    decode_numeric(&buf.get(1..).unwrap_or_default(), 10),
```

Comment:
Ohh that's the safe rusty way of indexing

Reply:
Empty string. Then `decode_numeric` will return an error.

This should never happen anyway. It is not a part of the fix, but a second "defensive" commit.